rllib/policy/sample_batch.py

```diff
@@ -59,19 +59,31 @@ class SampleBatch:
     def __init__(self, *args, **kwargs):
         """Constructs a sample batch (same params as dict constructor)."""
 
-        self._initial_inputs = kwargs.pop("_initial_inputs", {})
-
+        # Possible seq_lens (TxB or BxT) setup.
```

Comment:
Had to make these changes to SampleBatch (which are all backward compatible; not using these won't change a thing):
- make it possible to have a time-major SampleBatch with data format [TxBx... ] (instead of the currently only supported [Bx...]).
- time-major, there will be an additional seq_lens field indicating each batch-items seq-len.

Reply:
Does the time major thing apply to all fields, or just the state? I'm wondering how much speedup this actually has, since the transpose presumably isn't that expensive if you only need to do it once at train time (shouldn't need SampleBatch for inference at all). It would be cleaner to not add this new format to SampleBatch.

Reply:
The speedup is actually massive. I did a test vs the non-trajectory view code, simply switching the LSTM to time-major (even though the data was not!). The non-trajectory code was then much faster. However, after adding the data-transpose, it slowed down again to almost the same speed as before. So the bottom line is: Time-major is the main "thing" here in this PR, but in order for it to fully use its potential, the data must be collected already as time-major (in the SampleBatch).

Reply:
... as well as already be 0-padded. So we collect the data already exactly as the LSTM/RNN will need it later in the train pass.